### ISSUE TITLE
Incorrect Memory parsing for OSX

### ISSUE THREAD

**Issue by LotharSee** [DataDog/dd-agent#1338]
To get free/used memory on OSX, we call `top -l 1`, then look at the right line and extract the numbers.
https://github.com/DataDog/dd-agent/blob/master/checks/system/unix.py#L546

But we assume that these numbers unit is always M. Which is wrong since it seems to be G when it's too big.
A few output example:

The head of the `top -l 1` command

```
Processes: 247 total, 3 running, 3 stuck, 241 sleeping, 1204 threads
2015/01/28 13:58:50
Load Avg: 13.78, 15.70, 10.86
CPU usage: 4.12% user, 4.61% sys, 91.26% idle
SharedLibs: 14M resident, 7088K data, 0B linkedit.
MemRegions: 67846 total, 7816M resident, 67M private, 3414M shared.
PhysMem: 19G used (7041M wired), 11G unused.
VM: 680G vsize, 1066M framework vsize, 3214(0) swapins, 7182(0) swapouts.
Networks: packets: 1542992298/536G in, 1499758175/492G out.
Disks: 43995281/708G read, 125096660/3311G written.
```

The line we are parsing.

```
PhysMem: 19G used (7041M wired), 11G unused.
```

Here is it is only G. Another example.

```
PhysMem: 12G used (4355M wired), 3419M unused.
```

There is a mix of M and G.

Result: if `system.mem.used` or `system.mem.free` is bigger than 10G (let's say 12G), we report values 1000x times smaller (example: 12M).

Solution: also parse the unit to correct the value.

**Comment by MartinDatadog** [comment]
Customer reporting this: https://datadog.desk.com/agent/case/21001

**Comment by remh** [comment]
other solution: use psutil to get those metrics.

**Comment by remh** [comment]
Was fixed in 5.6.3 #2097
